packages/liveblocks-react/src/room.tsx

```diff
@@ -365,6 +367,44 @@ export function createRoomContext<
       })
     );
 
+    React.useEffect(() => {
+      async function handleCommentEvent(message: CommentsEventServerMsg) {
+        // TODO: Error handling
+        const info = await room.getThread({ threadId: message.threadId });
+
+        // If no thread info was returned (i.e., 404), we remove the thread and relevant inbox notifications from local cache.
+        if (!info) {
+          store.deleteThread(message.threadId);
+          return;
+        }
+        const { thread, inboxNotification } = info;
+
+        const existingThread = store.get().threads[message.threadId];
+
+        switch (message.type) {
+          case ServerMsgCode.COMMENT_EDITED:
+          case ServerMsgCode.THREAD_METADATA_UPDATED:
+          case ServerMsgCode.COMMENT_REACTION_ADDED:
+          case ServerMsgCode.COMMENT_REACTION_REMOVED:
+            // If the thread doesn't exist in the local cache, we do not update it with the server data as an optimistic update could have deleted the thread locally.
+            if (!existingThread) break;
```

Comment:
How is it possible to delete a thread with an optimistic update? In my mind, it can only happen after getting a 404 or a full refresh of the cache. Never with an optimistic update. What scenario do you have in mind?

Reply:
Hmmm. I realize the comment description isn't completely accurate. It should say optimistic update followed by successful mutation request.

I thought when we deleted the last comment of a thread (after mutation request is successful, we also deleted the thread from cache, but I checked the implementation of `useDeleteComment`, and that doesn't seem to be the case. Is that intentional or did we forget to account for that case?

https://github.com/liveblocks/liveblocks/assets/35807215/aec86f1e-f98d-4ea3-986c-a8544121bd14

Reply:
Pushed an update to delete a thread from cache if the deleted comment is the last undeleted comment in the thread. Here's [the commit link](https://github.com/liveblocks/liveblocks/pull/1374/commits/42b4e403944a957ba51a3cb722b86ad9f8aa8655) - let me know what you think!